agent_gateway/tools/snowflake_tools.py

```diff
@@ -383,20 +387,39 @@ def _prepare_analyst_request(self, prompt):
 
         return url, headers, data
 
-    def _process_message(self, response):
-        # ensure valid sql query is present in response
-        if response[1].get("type") != "sql":
-            return "Invalid Query"
-
-        # execute sql query
-        sql_query = response[1]["statement"]
-        gateway_logger.log(logging.DEBUG, f"Cortex Analyst SQL Query:{sql_query}")
-        table = self.connection.cursor().execute(sql_query).fetch_arrow_all()
+    def _process_analyst_message(self, response):
```

Comment:
We should look to add some isolated unit tests to ensure we're properly handling all of the analyst response combinations. It does not need to be done here; we can handle it in a separate PR.